### PR TITLE
feat(pie-docs): DSW-410 add "back to top" button

### PR DESCRIPTION
**Pull request by xander-marjoram**
## Describe your changes (can list changeset entries if preferable)
- [Added] - "Back to top" button to scroll to the top of the page

## Author Checklist (complete before requesting a review)
- [x] I have performed a self-review of my code
- [ ] If it is a core feature, I have added thorough tests
- [x] If it is a `PIE Docs` change, I have reviewed the Docs site preview
- [ ] If it is a component change, I have reviewed the Storybook preview
- [ ] If there are visual test updates, I have reviewed them properly before approving

### Reviewer 1
- [ ] If it is a `PIE Docs` change, I have reviewed the PR preview
- [ ] If there are visual test updates, I have reviewed them

### Reviewer 2
- [ ] If it is a `PIE Docs` change, I have reviewed the PR preview
- [ ] If there are visual test updates, I have reviewed them

### CODE REVIEW

File: apps/pie-docs/src/_includes/header.njk [hunk 1]
```diff
@@ -5,7 +5,7 @@
     {{ { name: 'close', attrs: { fill: toggleIconFill, height: '24', width: '24', class: 'u-iconFilled' } } | pieIconsSvg | safe }}
 </label>
 
-<header class="c-header" data-test-id="site_header">
+<header class="c-header" data-js="site-header" data-test-id="site_header">
```
Comment (jamieomaguire) [inline apps/pie-docs/src/_includes/header.njk:8]: NIT: the data-js attribute in default-layout uses camel case. Should we stick to one approach?

**Review by jamieomaguire — APPROVED** [review]

**Review by FayeCarter — APPROVED** [review]

File: apps/pie-docs/src/_includes/header.njk (hunk 1 above)
Comment (xander-marjoram) [reply apps/pie-docs/src/_includes/header.njk:8]: Updated all to kebab case

**Review by FayeCarter — APPROVED** [review]

**Review by jamieomaguire — APPROVED** [review]

File: apps/pie-docs/src/assets/js/scroll-to-top.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+    const scrollObserver = new IntersectionObserver((entries) => {
+        entries.forEach((entry) => {
+            calculateButtonVisibility(entry.isIntersecting);
+        });
+    });
+
+    if (siteHeader) {
+        scrollObserver.observe(siteHeader);
+    }
```
Comment (raoufswe) [inline apps/pie-docs/src/assets/js/scroll-to-top.js:17]: do you think we should unobserve / disconnect when the page unloads?

Comment (xander-marjoram) [reply apps/pie-docs/src/assets/js/scroll-to-top.js:17]: Interesting question, I think the browser's GC should handle it by default. I think it only becomes an issue when you have something like an SPA?

**Review by raoufswe — APPROVED** [review]

**Pull request merged**